src/generate/peripheral/accessor.rs

```diff
@@ -62,6 +65,21 @@ impl ToTokens for AccessType {
                     }
                 }
             }
+            Self::Ptr(Accessor::Reg(RegAccessor {
+                doc,
+                name,
+                ty,
+                offset,
+            })) => {
+                let offset = (*offset != 0).then(|| unsuffixed(*offset)).map(|o| quote!(.add(#o)));
+                quote! {
+                    #[doc = #doc]
+                    #[inline(always)]
+                    pub const fn #name(&self) -> #ty {
```

Comment:
Do we not want &#ty here? Same for the other accessors

Reply:
I think that, with raw access, peripherals return pointers to registers instead of a reference to one of the fields of their register block. Thus, instead of borrowing access, users are "owners" of these registers.